projects/common/src/telemetry/track/track.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import { UserTelemetryInternalService } from '../user-telemetry-internal.service';
+
+@Directive({
+  selector: '[htTrack]'
+})
+export class TrackDirective {
+  @Input('htTrack')
+  public name!: string;
+
+  public constructor(private readonly userTelemetryInternalService: UserTelemetryInternalService) {}
+
+  @HostListener('click', ['$event'])
```

Comment:
So how can we make this support events other than click? This is something we should consider now because adding that in the future will break usages.

Reply:
From a Product/Marketing perspective, hover and mouseover events are not very interesting. 'Click' and possibly 'Context menu click' would be worth tracking. If you are trying to optimize for UX teams, then we really need  'auto-tracking' based telemetry tools.

So, I was debating between keeping different directives for each event type vs one directive with multiple config options. IMO, multiple config options would be better from a usability perspective (for our current use case). 

We can support properties like `enableClickTracking, enableContextMenuTracking, etc` and for each type, we can have a host Listener binding that would emit only when the corresponding flag is enabled.

Of course, a host listener would still be attached for disabled events. So, it's not ideal as well. But it would be better than having multiple tracking directives on the same element especially during the initial cases.

Reply:
click, right click, enter/return - there are other relevant events. Why not just `htTrack="click" [trackEventName]="label"`? Using a host listener isn't a great fit here - see `EventBlockerComponent` - it had a similar requirement.

Reply:
lemme check

Reply:
Yeah, we can take an array of event types as an input (and assign a reasonable default).